### ISSUE TITLE
[Bug]: Importing from another route duplicates code

### ISSUE THREAD

**Issue by sergiodxa** [remix-run/remix#1143]
### Which Remix packages are impacted?

- [X] `remix` (Remix core)
- [ ] `create-remix`
- [ ] `@remix-run/architect`
- [ ] `@remix-run/cloudflare-workers`
- [ ] `@remix-run/dev`
- [ ] `@remix-run/express`
- [ ] `@remix-run/netlify`
- [ ] `@remix-run/node`
- [ ] `@remix-run/react`
- [ ] `@remix-run/serve`
- [ ] `@remix-run/server-runtime`
- [ ] `@remix-run/vercel`

### What version of Remix are you using?

Latest

### What version of Node are you using? Minimum supported version is 14.

Any version 

### Steps to Reproduce

- Create a context inside the root route and export it
- In the root component render the provider of the context
- In another root import that context and use it with useContext

### Expected Behavior

It should either not duplicate the code or throw an error if you import from another route.

### Actual Behavior

You don’t see any error or warning, and the useContext will not return the value the Provider received.

**Comment by geocine** [comment]
Have you tried creating the context outside of the `routes` folder and importing it accordingly to your routes?

**Comment by sergiodxa** [comment]
> Have you tried creating the context outside of the `routes` folder and importing it accordingly to your routes?

Yes, that’s what I always recommend everyone with this issue, I didn’t opened the issue because I have the problem but because a lot of people do this.

**Comment by brookslybrand** [comment]
Thank you for opening this issue, and our apologies we haven't gotten around to it yet!  

With the release of [React Router v7](https://remix.run/blog/react-router-v7) we are sun-setting continued development/maintenance on Remix v2.  If you have not already [upgraded to React Router v7](https://reactrouter.com/upgrading/remix), we recommend you do so. We've tried to make the upgrade process as smooth as possible with our [Future Flags](https://remix.run/docs/en/main/start/future-flags). We are now in the process of cleaning up outdated issues and pull requests to improve the overall hygiene of our repositories.  

We plan to continue to address 2 types of issues in Remix v2:

* Bugs that pose security concerns
* Bugs that prevent upgrading to React Router v7

If you believe this issue meets one of those criteria, please respond or [create a new issue](https://github.com/remix-run/remix/issues/new?template=bug_report.yml).

For all other issues, ongoing maintenance will be happening in React Router v7, so please reopen this issue [in that repo](https://github.com/remix-run/react-router/issues/new?template=bug_report.yml) with a new minimal reproduction against v7.

If you have any questions you can always reach out on [Discord](https://rmx.as/discord). Thanks again for providing feedback and helping us make our framework even better!
